apps/src/templates/amazonFutureEngineerEligibility/amazonFutureEngineerEligibility.jsx

```diff
@@ -1,51 +1,127 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import {FormGroup, Button} from 'react-bootstrap';
 import FieldGroup from '../../code-studio/pd/form_components/FieldGroup';
 import SchoolAutocompleteDropdownWithLabel from '@cdo/apps/templates/census2017/SchoolAutocompleteDropdownWithLabel';
 import AmazonFutureEngineerEligibilityForm from './amazonFutureEngineerEligibilityForm';
 import AmazonFutureEngineerAccountConfirmation from './amazonFutureEngineerAccountConfirmation';
+import {pegasus} from '@cdo/apps/lib/util/urlHelpers';
 
 const styles = {
   intro: {
     paddingBottom: 10
   }
 };
 
+const sessionStorageKey = 'AmazonFutureEngineerEligibility';
+
 export default class AmazonFutureEngineerEligibility extends React.Component {
+  static propTypes = {
+    signedIn: PropTypes.bool.isRequired
+  };
+
   constructor(props) {
     super(props);
 
+    let sessionEligibilityData =
+      JSON.parse(sessionStorage.getItem(sessionStorageKey)) || {};
+
     this.state = {
-      schoolEligible: null,
-      consent: false,
-      email: ''
+      formData: {
+        signedIn: this.props.signedIn,
+        schoolEligible:
+          'schoolEligible' in sessionEligibilityData
+            ? sessionEligibilityData.schoolEligible
+            : null,
+        consentAFE:
+          'consentAFE' in sessionEligibilityData
+            ? sessionEligibilityData.consentAFE
+            : false
+      }
     };
   }
 
-  handleChange = change => {
-    this.setState(change);
+  updateFormData = change => {
+    this.setState({formData: {...this.state.formData, ...change}});
```

Comment:
Have a lot of ugly combinations of objects throughout, would love input on cleaner ways to do this. I might just move `formData` out of its own object, but even if I do that I have other places in this code where I'm combining objects in an ugly way.

Reply:
If the state just holds a flat set of form fields at its top level, rather than using any nested fields, you should just be able to `setState` with changed fields each time?

Reply:
Yeah, I originally had `formData` and other info also stored in state, but slowly moved everything into `formData`. Can move it out, but just tagged you in another place where I'm doing what seems like gross stuff to combine objects if you have ideas on how to do that better.

Reply:
I personally don't find this to be pretty clean